### PR TITLE
#13763 task sorting fixed; existing task name search fixed

### PR DESCRIPTION
**Pull request by LonwoLonwo**
2021-09-03 06_24_18-DBeaver Ultimate 21 2 0 - comments

k1mUDvpeNq

### CODE REVIEW

**Comment by ShadelessFox** [comment]
We probably want to have this behavior instead of always keeping the folders on top:

EPBztKyXDz

File: plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/controls/ViewerColumnController.java [hunk 1]
```diff
@@ -786,6 +788,8 @@ public int compare(Viewer v, Object e1, Object e2)
                         result = -1;
                     } else if (value2 == null) {
                         result = 1;
+                    } else if (e1 instanceof DBTTask && e2 instanceof DBTTaskFolder) { // Folders should always be higher in the tree
+                        return 1;
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/controls/ViewerColumnController.java]: Why such generic control should know about tasks? This is a bad idea.

File: plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/DatabaseTasksTree.java [hunk 2]
```diff
@@ -568,6 +580,7 @@ private void addCheckedItem(TreeItem item, List<DBTTask> tasks) {
                 } else {
                     // Tasks
                     List<DBTTask> allTasksThisCategory = getSortedByCategoryTasks(parentCat);
+                    allTasksThisCategory.sort((o1, o2) -> o1.getName().compareToIgnoreCase(o2.getName()));
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/DatabaseTasksTree.java]:
```suggestion
                    allTasksThisCategory.sort(DBUtils.nameComparatorIgnoreCase());
```

File: plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/DatabaseTasksTree.java [hunk 3]
```diff
@@ -542,6 +553,7 @@ private void addCheckedItem(TreeItem item, List<DBTTask> tasks) {
                     thisFolderTasks = allTasks.stream().filter(task -> task.getTaskFolder() == taskFolder).collect(Collectors.toList());
                     // folderProject will be null in this case. It's ok
                 }
+                thisFolderTasks.sort((o1, o2) -> o1.getName().compareToIgnoreCase(o2.getName()));
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/DatabaseTasksTree.java]:
```suggestion
                thisFolderTasks.sort(DBUtils.nameComparatorIgnoreCase());
```

File: plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/DatabaseTasksTree.java [hunk 4]
```diff
@@ -335,9 +336,13 @@ void regroupTasks(ExpansionOptions options) {
...
                 // Add task folders as parent elements, task from these folders will be added in children list
                 if (!CommonUtils.isEmpty(allTasksFolders)) {
+                    allTasksFolders.sort((o1, o2) -> o1.getName().compareToIgnoreCase(o2.getName()));
                     rootObjects.addAll(allTasksFolders);
                 }
-                List<DBTTask> allTasksWithoutFolders = allTasks.stream().filter(task -> task.getTaskFolder() == null).collect(Collectors.toList());
+                List<DBTTask> allTasksWithoutFolders = allTasks.stream()
+                    .filter(task -> task.getTaskFolder() == null)
+                    .sorted((o1, o2) -> o1.getName().compareToIgnoreCase(o2.getName()))
+                    .collect(Collectors.toList());
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/DatabaseTasksTree.java]: Is sorting here essential? We can define a custom comparator for the viewer instead.

File: plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/DatabaseTasksTree.java [hunk 5]
```diff
@@ -514,9 +519,15 @@ private void addCheckedItem(TreeItem item, List<DBTTask> tasks) {
             if (parentElement instanceof DBPProject) {
                 DBPProject project = (DBPProject) parentElement;
                 // First add all tasks folders belonging to this project
-                children.addAll(allTasksFolders.stream().filter(taskFolder -> taskFolder.getProject() == parentElement).collect(Collectors.toList()));
+                children.addAll(allTasksFolders.stream()
+                    .filter(taskFolder -> taskFolder.getProject() == parentElement)
+                    .sorted((o1, o2) -> o1.getName().compareToIgnoreCase(o2.getName()))
+                    .collect(Collectors.toList()));
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/DatabaseTasksTree.java]: Is sorting here essential? We can define a custom comparator for the viewer instead.

File: plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/DatabaseTasksTree.java [hunk 6]
```diff
@@ -514,9 +519,15 @@ private void addCheckedItem(TreeItem item, List<DBTTask> tasks) {
...
                 // Then check all tasks belonging to this project without folder
-                List<DBTTask> thisProjectTasksWithoutFolder = allTasks.stream().filter(task -> task.getTaskFolder() == null && task.getProject() == parentElement).collect(Collectors.toList());
+                List<DBTTask> thisProjectTasksWithoutFolder = allTasks.stream()
+                    .filter(task -> task.getTaskFolder() == null && task.getProject() == parentElement)
+                    .sorted((o1, o2) -> o1.getName().compareToIgnoreCase(o2.getName()))
+                    .collect(Collectors.toList());
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/DatabaseTasksTree.java]: Is sorting here essential? We can define a custom comparator for the viewer instead.

File: plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/DatabaseTasksTree.java (hunk 3 above)
Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/DatabaseTasksTree.java]: Is sorting here essential? We can define a custom comparator for the viewer instead.

File: plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/DatabaseTasksTree.java (hunk 2 above)
Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/DatabaseTasksTree.java]: Is sorting here essential? We can define a custom comparator for the viewer instead.

**Review by ShadelessFox — CHANGES_REQUESTED** [review]

**Comment by serge-rider** [comment]
> We probably want to have this behavior instead of always keeping the folders on top:

No, we don't. We want folders on top

**Pull request closed without merging**